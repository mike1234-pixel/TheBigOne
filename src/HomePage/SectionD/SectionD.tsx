import React from "react";
import "./SectionD.scss";

const SectionD = () => {
  return (
    <div className="section-d">
      <div className="bg-video">
        <video className="bg-video__content" autoPlay muted loop>
          <source
            src={require("../../backgroundImages/video.mp4")}
            type="video/mp4"
          />
          Your browser is not supported
        </video>
      </div>
    </div>
  );
};
// video autoplay
export default SectionD;
