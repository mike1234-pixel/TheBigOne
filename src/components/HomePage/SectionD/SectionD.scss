.section-d {
  width: 100%;
  max-width: 100%;
  height: 75vh;
  position: relative;

  &-light {
    background: #fff;
  }
  &-dark {
    background: #000;
  }
  &-text {
    color: purple;
    z-index: 10;
    font-size: 40px;
    position: absolute;
    top: 65vh;
    left: 50vw;
    &:hover {
      color: red;
    }
    @media (max-width: 1300px) {
      font-size: 30px;
    }
    @media (max-width: 560px) {
      left: 40vw;
      top: 60vh;
    }
    @media (max-width: 340px) {
      font-size: 20px;
    }
  }
}

.bg-video {
  position: relative;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
  opacity: 0.15;
  overflow: hidden;

  &__content {
    height: 100%;
    width: 100%;
    object-fit: cover;
  }
}
