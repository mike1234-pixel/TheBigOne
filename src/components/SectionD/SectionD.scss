.section-d {
  width: 100vw;
  height: 75vh;
  background-color: rgb(20, 172, 108);
}
