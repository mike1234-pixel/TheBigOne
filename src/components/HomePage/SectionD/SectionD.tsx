import React from "react";
import "./SectionD.scss";
import PropTypes from "prop-types";

const SectionD = (props) => {
  return (
    <div className="section-d">
      <div className="bg-video">
        <video className="bg-video__content" autoPlay muted loop>
          <source
            src={require("../../../staticResources/backgroundImages/video.mp4")}
            type="video/mp4"
          />
          Your browser is not supported
        </video>
      </div>
    </div>
  );
};
// video autoplay
export default SectionD;

SectionD.propTypes = {
  darkMode: PropTypes.bool,
};
