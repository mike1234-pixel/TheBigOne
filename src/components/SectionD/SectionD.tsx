import React from "react";
import "./SectionD.scss";

const SectionD = () => {
  return <div className="section-d"></div>;
};

export default SectionD;
