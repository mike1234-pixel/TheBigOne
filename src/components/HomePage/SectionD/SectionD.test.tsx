import React from "react";
import SectionD from "./SectionD";
import renderer from "react-test-renderer";

// snapshot test

it("renders correctly", () => {
  const tree = renderer.create(<SectionD />).toJSON();
  expect(tree).toMatchSnapshot();
});
