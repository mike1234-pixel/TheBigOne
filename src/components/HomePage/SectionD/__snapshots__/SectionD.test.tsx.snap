// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="section-d"
>
  <div
    className="bg-video"
  >
    <video
      autoPlay={true}
      className="bg-video__content"
      loop={true}
      muted={true}
    >
      <source
        src="video.mp4"
        type="video/mp4"
      />
      Your browser is not supported
    </video>
  </div>
</div>
`;
